neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/data/im/BoundInMemoryBackend.java

```diff
@@ -0,0 +1,295 @@
+                    .averageKeySize(Sizes.ID)
+                    .averageValueSize(Sizes.FEATURE)
+                    .keyMarshaller(new BeanMarshaller<>(SERIALIZER_FACTORY.forId()))
+                    .valueMarshaller(new BeanMarshaller<>(SERIALIZER_FACTORY.forSingleFeature()))
+                    .create();
+
+            instances = ChronicleMapBuilder.of(Id.class, ClassBean.class)
+                    .name("bound/instances")
+                    .entries(Sizes.ENTRIES)
+                    .averageKeySize(Sizes.ID)
+                    .averageValueSize(Sizes.CLASS)
+                    .keyMarshaller(new BeanMarshaller<>(SERIALIZER_FACTORY.forId()))
+                    .valueMarshaller(new BeanMarshaller<>(SERIALIZER_FACTORY.forClass()))
+                    .create();
+
+            features = ChronicleMapBuilder.of(SingleFeatureBean.class, Object.class)
+                    .name("bound/features")
+                    .entries(Sizes.ENTRIES)
+                    .averageKeySize(Sizes.FEATURE)
+                    .averageValueSize(Sizes.FEATURE_VALUE)
+                    .keyMarshaller(new BeanMarshaller<>(SERIALIZER_FACTORY.forSingleFeature()))
+                    .create();
+
+            featuresById = new HashMap<>();
+        }
+
+        /**
+         * Cleans the data related to the specified {@code id}, and closes every maps if necessary.
+         *
+         * @param id the identifier of the data to clean
+         */
+        public void close(Id id) {
+            // Remove the container from the id if present (accessible only by the id)
+            containers.remove(id);
+
+            // Unregister the current back-end and clear all features associated with the id
+            Set<Integer> relatedFeatures = featuresById.remove(id);
+            if (nonNull(relatedFeatures)) {
+                relatedFeatures.forEach(n -> features.remove(SingleFeatureBean.of(id, n)));
+            }
+
+            // Cleans all shared in-memory maps: they will no longer be used
+            if (COUNTER.get() == 0L) {
+                Log.debug("Cleaning BoundTransientBackend");
+
+                containers.clear();
+                containers.close();
+
+                instances.clear();
+                instances.close();
+
+                features.clear();
+                features.close();
+
+                featuresById.clear();
+            }
+        }
+
+        /**
+         * Checks whether the maps are closed.
+         *
+         * @return {@code true} if the maps are closed, or not initialized yet.
+         */
+        public boolean isClosed() {
+            // If the 'containers' map is closed, we consider that all maps are closed
+            return isNull(containers) || !containers.isOpen();
+        }
+
+        /**
+         * The initialization-on-demand holder of the singleton of this class.
+         */
+        @Static
+        private static final class Holder {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)